### DISCUSSION TITLE
Use externally hosted MariaDB (on Home Assistant)

### DISCUSSION THREAD

**Discussion by lits101** [bassmaster187/TeslaLogger#1276] [Q&A]
Is it possible to configure the docker-compose.yaml to use an externally hosted database. I have MariaDB running as an add-on in Home Assistant, and I host a lot of things on there because it's backed up regularly and then backed up to Google Drive automatically. 

I've specified environment variables:
      - MYSQL_USER=teslalogger
      - MYSQL_PASSWORD=xxxxxx
      - MYSQL_DATABASE=teslalogger
      - MYSQL_HOST=192.168.20.x

This doesn't seem to have any effect, the main teslalogger container is still trying to access the 'database' container, which I've commented out.

**Comment by bassmaster187** [comment]
copy App.config in Teslalogger.exe folder and rename it to Teslalogger.exe.config
https://github.com/bassmaster187/TeslaLogger/blob/master/TeslaLogger/App.config

There is a tag called DBConnectionstring you can use to connect to a different database.

Something like that should work:
```

...
<setting name="DBConnectionstring" serializeAs="String">
        <value>Server=192.168.20.x;Database=teslalogger;Uid=root;Password=teslalogger;CharSet=utf8mb4;</value>
      </setting>
....


```

**Reply by lits101** [reply]
Ah yes, I see the setting in the file. I've specified my variables and restarted the container but it's still attempting to connect to the default database.

**Reply by bassmaster187** [reply]
it must work, because I'm using different databases for testing and developing. Make sure the name is:
`
TeslaLogger.exe.config
`
It is case sensitive! Make sure you didn't forget to close tags or have any other syntax errors in your settings. There are also some limitations for special characters for name / password. Make sure you're using just letters and numbers and no special characters...

**Reply by lits101** [reply]
I've copied the TeslaLogger/App.config file to TeslaLogger/TeslaLogger.exe.config. 
```
-rw-r--r--  1 pi   pi   5.0K May  1 07:27 App.config
...
-rw-r--r--  1 pi   pi   5.0K May  1 08:33 TeslaLogger.exe.config
```

Here's the codeblock from my new file TeslaLogger/TeslaLogger.exe.config:
```
      <setting name="DBConnectionstring" serializeAs="String">
        <value>Server=192.168.20.246;Database=teslalogger;Uid=teslalogger;Password=xxsuitablepasswordxx;CharSet=utf8mb4;</value>
      </setting>
```

I've restarted the container stack. No effect. 

The log still shows: 
`DBConnectionstring: Server=database;Database=teslalogger;Uid=root;Password=teslalogger;CharSet=utf8mb4;`

**Comment by l4n-socrates** [comment]
Same problem. I wonder, where are the actual settings are stored. Becouse, the App.config looks like the template. So, in my understanding, when i copied and rename it to TeslaLogger.exe.config, all my settings i made are overriden by the defaults? I also dont have a folder "Teslalogger.exe", its just ../TeslaLogger. So, should ist name the file TeslaLogger.config?

Or is it maybe in /TeslaLogger/bin ?
Becouse, there are also the file TeslaLogger.exe and other files like TeslaFi-Import.exe.config
This is also the path, wich is mounted in the docker-compuse.yml to the internal /etc/teslalogger
./TeslaLogger/bin:/etc/teslalogger
